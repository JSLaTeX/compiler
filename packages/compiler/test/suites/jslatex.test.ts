import * as path from 'node:path';
import { expect, test, describe } from 'vitest';
import { outdent } from 'outdent';
import { join } from 'desm';
import { compileJsLatex, compileJsLatexFile } from '~/index.js';

const fixturesDir = join(import.meta.url, '../fixtures');

test('compiles JSLaTeX', async () => {
	const jsLatex = outdent.string(String.raw`
		\documentclass{article}
		<?= "Hello from EJS!" ?>
	`);

	const latex = await compileJsLatex({ latex: jsLatex });

	expect(latex).toEqual(
		outdent.string(String.raw`
			\documentclass{article}
			Hello from EJS!
		`)
	);
});

test('compiles JSLaTeX with ramda', async () => {
	const jsLatex = outdent.string(String.raw`
		\documentclass{article}
		<?= R.sum(R.range(1, 11)) ?>
	`);

	const latex = await compileJsLatex({ latex: jsLatex });

	expect(latex).toEqual(
		outdent.string(String.raw`
			\documentclass{article}
			55
		`)
	);
});

describe('compiles fixtures', () => {
	test('chem.tex', async () => {
		expect(
			await compileJsLatexFile({ filePath: path.join(fixturesDir, 'chem.tex') })
		).toMatchSnapshot();
	});

	test('cowsay.tex', async () => {
		expect(
			await compileJsLatexFile({
				filePath: path.join(fixturesDir, 'cowsay/cowsay.tex'),
				projectBaseUrl: path.join(fixturesDir, 'cowsay/'),
			})
		).toMatchSnapshot();
	});
});
