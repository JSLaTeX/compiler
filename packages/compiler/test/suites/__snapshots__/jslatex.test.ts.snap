// Vitest Snapshot v1

exports[`compiles fixtures > chem.tex 1`] = `
"\\\\documentclass{article}
\\\\usepackage[margin=1in]{geometry}
\\\\usepackage{float}
\\\\usepackage{tabularx}
\\\\usepackage{ragged2e}
\\\\usepackage[version=4]{mhchem}



\\\\begin{document}
	\\\\begin{table}[H]
		\\\\begin{tabularx}{\\\\linewidth}{|
				>{\\\\RaggedRight}X|
				>{\\\\RaggedRight}X|
				>{\\\\RaggedRight}X|
				>{\\\\RaggedRight}X|
				>{\\\\RaggedRight}X|
				>{\\\\RaggedRight}X|
			}
			\\\\hline
			Trial \\\\#
			& Volume of \\\\ce{CuSO4} /mL
			& Baseline temperature /$^{\\\\circ}$C
			& Time of \\\\ce{Zn} addition /s
			& Equation of cooling line
			& R value
			\\\\\\\\\\\\hline
			
				1
				& 20
				& 24.4
				& 72
				& -0.042 - 0.042$x$
				& -0.992
				\\\\\\\\\\\\hline
			
				2
				& 20
				& 24.2
				& 132
				& -0.052 - 0.052$x$
				& -0.959
				\\\\\\\\\\\\hline
			
				3
				& 20
				& 24.2
				& 60
				& -0.086 - 0.086$x$
				& -0.993
				\\\\\\\\\\\\hline
			
				4
				& 20
				& 24.8
				& 72
				& -0.005 - 0.005$x$
				& -0.951
				\\\\\\\\\\\\hline
			
				5
				& 20
				& 24.6
				& 96
				& -0.042 - 0.042$x$
				& -0.996
				\\\\\\\\\\\\hline
			
				6
				& 20
				& 23.1
				& 14
				& 0.006 + 0.006$x$
				& 0.952
				\\\\\\\\\\\\hline
			
				7
				& 20
				& 23.8
				& 15
				& -0.060 - 0.060$x$
				& -0.897
				\\\\\\\\\\\\hline
			
				8
				& 20
				& 24.6
				& 96
				& -0.029 - 0.029$x$
				& -0.997
				\\\\\\\\\\\\hline
			
		\\\\end{tabularx}
	\\\\end{table}
\\\\end{document}
"
`;

exports[`compiles fixtures > cowsay.tex 1`] = `
"\\\\documentclass{article}



\\\\begin{document}

\\\\begin{verbatim}
	 ________
< LaTeX! >
 --------
        \\\\   ^__^
         \\\\  (oo)\\\\_______
            (__)\\\\       )\\\\/\\\\
                ||----w |
                ||     ||
\\\\end{verbatim}

\\\\end{document}
"
`;
